docs/Examples.md

```diff
@@ -87,6 +87,183 @@ $graphServiceClient = GraphServiceClient::createWithRequestAdapter($requestAdapt
+**`{tenantId}-{clientId}`** and for those that require a signed in user (delegated permissions), the cache key will be
+**`{tenantId}-{clientId}-{userId}`**.
+
+To retrieve the access token persisted to your custom cache for a particular user's/application's `TokenRequestContext`:
+```php
+
+$accessToken = $customCache->getAccessToken($tokenRequestContext->getCacheKey());
+
+```
+
+### Initializing a GraphServiceClient with an access token
+
+For applications that already have built-in mechanisms to fetch and refresh access tokens, the SDK supports passing these tokens to a `GraphServiceClient` by initializing
+a client using an [`AccessTokenCache`](https://github.com/microsoft/kiota-authentication-phpleague-php/blob/dev/src/Cache/AccessTokenCache.php) interface implementation.
+
+The SDK provides a built-in implementation of this interface via an [`InMemoryAccessTokenCache`](https://github.com/microsoft/kiota-authentication-phpleague-php/blob/dev/src/Cache/InMemoryAccessTokenCache.php).
+
+This is also useful when re-using a previously retrieved access token for a signed-in user during a previous request.
+
+The SDK will check the cache for a valid token before considering requesting a new token. If the provided token is expired
+and a refresh token is present, the access token will be refreshed and persisted to the cache. If no refresh token is provided, the SDK requests attempts to retrieve a new access token and persists it to the cache. In cases where a signed in user is present, e.g. authorization_code OAuth flows, the new token request will most likely fail because no valid `authorization_code` will be present meaning the user has to sign in again.
+
+*Using the `InMemoryAccessTokenCache`*:
+The in-memory cache can be hydrated/initialised using the `TokenRequestContext` and a PHPLeague [`AccessToken`](https://github.com/thephpleague/oauth2-client/blob/master/src/Token/AccessToken.php) object for a user/application:
+
+```php
+
+use Microsoft\Kiota\Authentication\Cache\InMemoryAccessTokenCache;
+use Microsoft\Graph\GraphServiceClient;
+use Microsoft\Graph\Core\Authentication\GraphPhpLeagueAccessTokenProvider;
+use Microsoft\Graph\Core\Authentication\GraphPhpLeagueAuthenticationProvider;
+use League\OAuth2\Client\Token\AccessToken;
+
+
+$cache = new InMemoryAccessTokenCache(
+    $tokenRequestContext,
+    new AccessToken(
+        [
+            'access_token' => $accessToken,
+            'refresh_token' => $refreshToken,
+            'expires' => 1
+        ]
+    )
+);
+
+$graphServiceClient = GraphServiceClient::createWithAuthenticationProvider(
+    GraphPhpLeagueAuthenticationProvider::createWithAccessTokenProvider(
+        GraphPhpLeagueAccessTokenProvider::createWithCache(
+            $cache,
+            $tokenRequestContext,
+            $scopes
+        )
+    )
+);
+
+```
+
+To initialize the cache with multiple TokenRequestContext-AccessToken pairs, `withToken` can be used:
```

Comment:
Highlighting which scenarios this serves would be ideal (I assume it's for an app doing requests on behalf of multiple users, like a syncing app?)

Reply:
Thanks for this. Updated the section to reflect a generic scenario.